python_modules/dagster/dagster/_core/definitions/input.py

```diff
@@ -323,6 +323,19 @@ def combine_with_inferred(self, inferred: InferredInputProps) -> "InputDefinitio
             input_manager_key=self._input_manager_key,
         )
 
+    def with_dagster_type(self, dagster_type: DagsterType) -> "InputDefinition":
+        return InputDefinition(
+            name=self.name,
+            dagster_type=dagster_type,
+            description=self.description,
+            default_value=self.default_value if self.has_default_value else NoValueSentinel,
+            root_manager_key=self._root_manager_key,
+            metadata=self._metadata,
+            asset_key=self._asset_key,
+            asset_partitions=self._asset_partitions_fn,
+            input_manager_key=self._input_manager_key,
+        )
```

Comment:
did you consider re-using `combine_with_inferred` either as the call or in this impl? Main motivation would be to reduce spooky copy-all-the-properties-over call sites.